### PR TITLE
Don't test reverse execution in Simics 7

### PR DESCRIPTION
**Pull request by aragners**
The reverse exectuion feature is getting removed in 7.

### CODE REVIEW

File: test/1.4/lib/T_signal_templates.py [hunk 1]
```diff
@@ -53,12 +53,14 @@ def signal_lower(self):
...
 stest.expect_equal(stub.level, 1)
 
-# reverse execution doesn't cause signal pulsing
-transitions = stub.transitions
-simics.SIM_run_command('enable-reverse-execution')
-simics.SIM_run_command('continue 1000')
-simics.SIM_run_command('reverse')
-stest.expect_equal(stub.transitions, transitions)
+# Reverse execution is removed in Simics 7
+if int(simics.SIM_version_major()) < 7:
```
Comment (lwaern-intel) [inline test/1.4/lib/T_signal_templates.py:57]: This'll crash if the version in play is 4.8, won't it? Not that I expect that ever happens, but still, for the sake of principle I'd suggest:
`if simics.SIM_version_major() == '4.8' or int(simics.SIM_version_major()) < 7:`

**Review by lwaern-intel — CHANGES_REQUESTED** [review]

File: test/1.4/lib/T_signal_templates.py (hunk 1 above)
Comment (aragners) [reply test/1.4/lib/T_signal_templates.py:57]: Would  `int(simics.SIM_version_major().split(".")[0]) < 7` be okay? Less clean but works with outdated Simics versions.

Comment (aragners) [reply test/1.4/lib/T_signal_templates.py:57]: When I think more about it I think DML 1.4 was added in 6, maybe with some tech preview in 5. I don't think it's worth adding the extra ugliness for supporting this test to be allowed to run in old Simics versions with "x.y" major versions.

Comment (lwaern-intel) [reply test/1.4/lib/T_signal_templates.py:57]: DMLC in principle still supports 4.8. But yes, this is nit-picky, and I only brought it out of principle's sake. It's not a blocker.

Comment (lwaern-intel) [reply test/1.4/lib/T_signal_templates.py:57]: Actually... rather, what we support is _API_ version 4.8. I'm not sure what majors of Simics proper we support. This makes this more shaky.

Comment (aragners) [reply test/1.4/lib/T_signal_templates.py]: I changed the check to see if rev-execution class is available or not.

**Comment by syssimics** [comment]
Verification [#12603417](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=12603417): fail

**Comment by syssimics** [comment]
Verification [#12603867](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=12603867): pass

**Review by lwaern-intel — APPROVED** [review]
The only concern was if this was somehow very slow, but it's not.

File: test/1.4/lib/T_signal_templates.py [hunk 2]
```diff
@@ -53,12 +53,14 @@ def signal_lower(self):
...
-simics.SIM_run_command('reverse')
-stest.expect_equal(stub.transitions, transitions)
+# Reverse execution is removed in Simics 7
+if 'rev-execution' in simics.SIM_get_all_classes():
+    # reverse execution doesn't cause signal pulsing
+    transitions = stub.transitions
+    simics.SIM_run_command('enable-reverse-execution')
+    simics.SIM_run_command('continue 1000')
+    simics.SIM_run_command('reverse')
+    stest.expect_equal(stub.transitions, transitions)
```
Comment (mandolaerik) [inline test/1.4/lib/T_signal_templates.py:63]: Better to rewrite the test to use the 7 equivalent of revexec (snapshots, right?)

FWIW, I think the purpose of the test is to exercise the following logic in `signal_connect.set`, in `1.4/utility.dml`:
```
        local bool hotplug = SIM_object_is_configured(dev.obj)
            && !SIM_is_restoring_state(dev.obj);
```

Comment (aragners) [reply test/1.4/lib/T_signal_templates.py:63]: I agree, but as snapshots API is not finalized yet I think we can wait for that. Posted https://jira.devtools.intel.com/browse/SIMICS-21789 to fix this.

**Pull request merged**